### PR TITLE
2253: Material table - filters reset on change tab

### LINKED ISSUE #2253: Material table - filters reset on change tab
Our material tables aren't persistent.  They are only "alive" when their tab is alive.

This means that when we change to another tab, and then return, any filters have been lost.

I _think_ we should get around this by storing the state of the filters for each table in the parent `SupportPanel` class (unless there is a better solution).

So, for example, `planning-assets` will have props for `filters` and `setFilters()`.   When it loads, if `filters` is empty, it will generate its columns object, and store either the columns or the currently applied filters in the parent via `setFilters`.  When the player changes tab, and comes back again, it will receive the stored value of `fliters` - and the original set of filters will be restored.

### PR DESCRIPTION
**Pull request by lebaphi**
Fixes #2253

### CODE REVIEW

**Comment by IanMayo** [comment]
Oops @lebaphi  - we've both worked on this issue.  I have this PR for it:
https://github.com/serge-web/serge/pull/2254

I _think_ we've both taken the same strategy. Would you mind checking mine, please?

**Comment by lebaphi** [comment]
> Oops @lebaphi - we've both worked on this issue. I have this PR for it: #2254
> 
> I _think_ we've both taken the same strategy. Would you mind checking mine, please?

Hi @IanMayo , yes we've both taken the same strategy 🤣

**Comment by lebaphi** [comment]
But I have updated the fix for the subtypes filter on `custom-filter-row.tsx`

**Comment by IanMayo** [comment]
> Hi @IanMayo , yes we've both taken the same strategy 🤣

Yours is neater.  I'll pull over a couple of my extra changes and merge it.

**Comment by IanMayo** [comment]
> if (asset.attributes && ((filter.length && filter.includes(asset.platformTypeId)) || !filter.length)) {

Could you explain how the above line works please?

**Comment by lebaphi** [comment]
> > if (asset.attributes && ((filter.length && filter.includes(asset.platformTypeId)) || !filter.length)) {
> 
> Could you explain how the above line works please?

hi @IanMayo, that means we build a subTypes list base on corresponding selected platformType or build all subTypes if no platformType is selected

File: packages/components/src/local/organisms/planning-assets/helpers/custom-filter-row.tsx [hunk 1]
```diff
@@ -24,7 +24,7 @@ const CustomFilterRow: React.FC<CustomFilterRowProps> = (props): React.ReactElem
         return
       }
       force.assets.forEach((asset: Asset) => {
-        if (asset.attributes && ((filter.length && filter.includes(asset.platformTypeId)) || !filter.length)) {
+        if (asset.attributes && (!filter.length || filter.includes(asset.platformTypeId))) {
```
Comment (IanMayo) [inline packages/components/src/local/organisms/planning-assets/helpers/custom-filter-row.tsx:27]: you ok with this refactoring @lebaphi ?

**Review by IanMayo — APPROVED** [review]

**Pull request merged**